h5py/h5t.pyx

```diff
@@ -1467,7 +1501,7 @@ cdef TypeReferenceID _c_ref(object refclass):
     raise TypeError("Unrecognized reference code")
 
 
-cpdef TypeID py_create(object dtype_in, bint logical=0):
+cpdef TypeID py_create(object dtype_in, bint logical=0, bint aligned=0):
```

Comment:
I think it's there so that it can take strings/other types that can be converted to a dtype (although I'm still not clear about the c/cython-level dtype code works, as dtypes from what I've read of the numpy docs are python level).